python/mpt/src/mpt/utils.cairo

```diff
@@ -123,3 +124,54 @@ func deserialize_to_internal_node{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}
     );
     return result;
 }
+
+// @notice Checks if a branch node is valid.
+// A branch node is valid if it has at least two non-null subnodes.
+// @dev Raises an error if the branch node is invalid.
+// @param node The branch node to check.
+func check_branch_node(node: BranchNode) {
+    alloc_locals;
+
+    local first_non_null_index;
+    local second_non_null_index;
+    let subnodes_ptr = cast(node.value.subnodes.value, felt*);
+
+    %{ find_two_non_null_subnodes %}
+
+    // Check that the first subnode is not None and not empty
+    tempvar x = Extended(cast(subnodes_ptr[first_non_null_index], ExtendedEnum*));
+
+    if (cast(x.value, felt) == 0) {
+        raise('ValueError');
+    }
+    // Case 1: subnode is a digest
+    if (cast(x.value.bytes.value, felt) != 0) {
+        if (x.value.bytes.value.len == 0) {
+            raise('ValueError');
```

Comment:
looks like only this case is covered by the test

Reply:
i think it's just the number of examples

Reply:
ah no, it's that we can't serialize empty arrays in this context

Reply:
example:

```python
def invalid_branch_node_strategy():
    return st.builds(
        BranchNode,
        subnodes=st.one_of(
            st.integers(min_value=0, max_value=2**256 - 1).map(
                lambda x: x.to_bytes(32, "little")
            ),
            st.from_type(LeafNode).map(lambda x: encode_internal_node(x)),
            st.from_type(ExtensionNode).map(lambda x: encode_internal_node(x)),
            st.none(),
            st.just([]),
        ).map(lambda subnode: tuple([b""] * 13 + [] + [] + [subnode])),
        value=st.just(b""),
    )
```

Just results in a tuple of subnodes of len < 16 which won't cause us to compare an item to an empty list.

Reply:
ok yeah i get it 👍

Reply:
Trying with [[]] instead of []

Reply:
now this one is no longer covered 😢

Reply:
`map(lambda subnode: tuple([b""] + [[]] + [b""] * 13)),`

I think this is what you want to have as strat to cover things properly? simply two null nodes, each one of different type? 😅